src/tools/sssctl/sssctl_user_checks.c

```diff
@@ -246,7 +248,13 @@ errno_t sssctl_user_checks(struct sss_cmdline *cmdline,
     if ( strncmp(action, "auth", 4)== 0 ) {
         fprintf(stdout, _("testing pam_authenticate\n\n"));
         ret = pam_authenticate(pamh, 0);
-        fprintf(stderr, _("pam_authenticate: %s\n\n"), pam_strerror(pamh, ret));
+        pret = pam_get_item(pamh, PAM_USER, (const void **) &pam_user);
+        if (pret != PAM_SUCCESS) {
+            fprintf(stderr, _("pam_get_item failed: %s\n"), pam_strerror(pamh,
+                                                                         pret));
+        }
+        fprintf(stderr, _("pam_authenticate for user [%s]: %s\n\n"), pam_user,
+                                                       pam_strerror(pamh, ret));
```

Comment:
The result of pam_authenticate() should still be printed even if the user name cannot be read. I set pam_user to a fixed string in the error case to have a reliable output.